### PR TITLE
Implement follow up referrals

### PR DESCRIPTION
**Pull request by cozej4**
@ellykits Please review this followup referrals implementation

### CODE REVIEW

File: opensrp-chw-referral/src/main/java/org/smartregister/chw/referral/TestUtil.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package org.smartregister.chw.referral;
 
 public class TestUtil {
-    public static int addTwoNumbers(int a, int b){
-        return a + b ;
+    public static int addTwoNumbers(int a, int b) {
+        return a + b;
     }
 }
```
Comment (ellykits) [inline opensrp-chw-referral/src/main/java/org/smartregister/chw/referral/TestUtil.java]: Kindly delete this class it was just for testing coveralls configuration

Comment (ellykits) [reply opensrp-chw-referral/src/main/java/org/smartregister/chw/referral/TestUtil.java]: Also delete the coresponding test class

File: opensrp-chw-referral/src/test/java/org/smartregister/chw/referral/ExampleUnitTest.java [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 import org.junit.Test;
```
Comment (ellykits) [inline opensrp-chw-referral/src/test/java/org/smartregister/chw/referral/ExampleUnitTest.java]: Delete this class please

File: opensrp-chw-referral/src/test/java/org/smartregister/chw/referral/custom_views/BaseReferralFloatingMenuTest.java [hunk 3]
```diff
@@ -6,20 +6,22 @@
 import org.mockito.Mockito;
 import org.powermock.core.classloader.annotations.PrepareForTest;
 import org.powermock.modules.junit4.rule.PowerMockRule;
+
 @PrepareForTest(BaseReferralFloatingMenu.class)
 public class BaseReferralFloatingMenuTest {
```
Comment (ellykits) [inline opensrp-chw-referral/src/test/java/org/smartregister/chw/referral/custom_views/BaseReferralFloatingMenuTest.java:11]: Could also test the core functionality of this class? What has been added so far is good but to avoid regressions, the reported test coverage should reflect the actual unit test of the functionality and the edge cases.

File: opensrp-chw-referral/src/test/java/org/smartregister/chw/referral/model/BaseReferralFollowupModelTest.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                        Assert.assertEquals(model.referralFollowupObject.getChwFollowupDate(), jsonObject1.optString("value"));
+                        break;
+                    default:
+                        Timber.i("field not found");
+                        break;
+                }
+            }
+
+        } catch (JSONException e) {
+            e.printStackTrace();
```
Comment (ellykits) [inline opensrp-chw-referral/src/test/java/org/smartregister/chw/referral/model/BaseReferralFollowupModelTest.java:86]: Use ```Timber```

**Review by ellykits — CHANGES_REQUESTED** [review]

**Review by ellykits — DISMISSED** [review]

**Review by ellykits — APPROVED** [review]
